numpyro/contrib/funsor/infer_util.py

```diff
@@ -141,18 +141,18 @@ def model(*args, **kwargs):
                 log_prob = scale * log_prob
 
             dim_to_name = site["infer"]["dim_to_name"]
-            log_prob = funsor.to_funsor(log_prob, output=funsor.reals(), dim_to_name=dim_to_name)
+            log_prob_factor = funsor.to_funsor(log_prob, output=funsor.reals(), dim_to_name=dim_to_name)
 
             time_dim = None
             for dim, name in dim_to_name.items():
                 if name.startswith("_time"):
-                    time_dim = funsor.Variable(name, funsor.domains.bint(site["value"].shape[dim]))
-                    time_to_factors[time_dim].append(log_prob)
+                    time_dim = funsor.Variable(name, funsor.domains.bint(log_prob.shape[dim]))
```

Comment:
@eb8680 I think this is a bug so I fixed it here. This makes enum fails when we have sites with `event_shape != ()`. I already add tests for the current examples (in those examples, `log_det` terms have `event_shape = (0,)`). What do you think?